pyriemann/utils/covariance.py

```diff
@@ -28,6 +28,48 @@ def _mcd(X):
     return C
 
 
+def schaefer_strimmer_cov(X):
+    """Schaefer-Strimmer covariance estimator
+
+    Shrinkage estimator using method from [1]:
+    .. math::
+            \hat{\Sigma} = (1 - \gamma)\Sigma_{scm} + \gamma T
+
+    where :math:`T` is the diagonal target matrix:
+    .. math::
+            T_{i,j} = \{ \Sigma_{scm}^{ii} \text{if} i = j, 0 \text{otherwise} \}
+    Note that the optimal :math:`\gamma` is estimate by the authors' method.
+
+    :param X: Signal matrix, Nchannels X Nsamples
+
+    :returns: Schaefer-Strimmer shrinkage covariance matrix, Nchannels X Nchannels
+
+    References
+    ----------
+    [1] Schafer, J., and K. Strimmer. 2005. A shrinkage approach to
+    large-scale covariance estimation and implications for functional
+    genomics. Statist. Appl. Genet. Mol. Biol. 4:32.
+    http://doi.org/10.2202/1544-6115.1175
+    """
+    Ne, Ns = X.shape[0], X.shape[1]
+    C_scm = numpy.cov(X, ddof=0)
+    X_c = X - numpy.tile(X.mean(axis=1), [Ns, 1]).T
+
+    # Compute optimal gamma, the weigthing between SCM and srinkage estimator
+    R = Ns / (Ns - 1.0) * numpy.corrcoef(X)
+    var_R = numpy.zeros(shape=(Ne, Ne))
+    for s in range(Ns):
+        for i in range(Ne):
+            for j in range(Ne):
+                var_R[i, j] += (X_c[i, s] * X_c[j, s] - C_scm[i, j]) ** 2
```

Comment:
I suspect einsum can help here.

also why do you use numpy. and not np. everywhere?